infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/UniqueNamesProvisioner.java

```diff
@@ -48,15 +59,38 @@ public void provision(T k8sEnv, RuntimeIdentity identity) throws InfrastructureE
 
     TracingTags.WORKSPACE_ID.set(workspaceId);
 
-    final Set<Pod> pods = new HashSet<>(k8sEnv.getPods().values());
-    k8sEnv.getPods().clear();
-    for (Pod pod : pods) {
+    final Map<String, ConfigMap> configMaps = k8sEnv.getConfigMaps();
+    Map<String, String> configMapNameTranslation = new HashMap<>();
+    for (ConfigMap configMap : configMaps.values()) {
+      final String originalName = configMap.getMetadata().getName();
+      putLabel(configMap.getMetadata(), Constants.CHE_ORIGINAL_NAME_LABEL, originalName);
+      final String uniqueName = Names.uniqueResourceName(originalName, workspaceId);
+      configMap.getMetadata().setName(uniqueName);
+      configMapNameTranslation.put(originalName, uniqueName);
+    }
+
+    final Collection<PodData> podData = k8sEnv.getPodsData().values();
+    for (PodData pod : podData) {
       final ObjectMeta podMeta = pod.getMetadata();
-      putLabel(pod, Constants.CHE_ORIGINAL_NAME_LABEL, podMeta.getName());
-      final String podName = Names.uniquePodName(podMeta.getName(), workspaceId);
+      putLabel(podMeta, Constants.CHE_ORIGINAL_NAME_LABEL, podMeta.getName());
+      final String podName = Names.uniqueResourceName(podMeta.getName(), workspaceId);
       podMeta.setName(podName);
```

Comment:
I'm not sure, I believe the behavior should be unchanged. The only difference is in method signatures (use meta instead of pod itself to support deployments) and renaming `#uniquePodName() -> #uniqueResourceName()` since it no longer only modifies pods.